server/spec/requests/foci_spec.rb

```diff
@@ -0,0 +1,346 @@
+    AllowlistDomain.create(email_domain: "dreamworks.com", usertype: "company representative", company_id: 3)
+    AllowlistEmail.create(email: "a@disney.com", usertype: "company representative", company_id: 1)
+    AllowlistEmail.create(email: "b@disney.com", usertype: "company representative", company_id: 1)
+    AllowlistEmail.create(email: "a@waltdisney.com", usertype: "company representative", company_id: 1)
+    AllowlistEmail.create(email: "c@dreamworks.com", usertype: "company representative", company_id: 3)
+    AllowlistEmail.create(email: "d@dreamworks.com", usertype: "company representative", company_id: 3)
+    @admin = User.find(1)
+    @student = User.create(id:3, firstname: "j", lastname: "s", email: "js@student.com", password: "pw", usertype: "student")
+    @student.email_activate
+    @rep = User.create(id: 2, firstname: "a", lastname: "s", email: "a@disney.com", password: "pw", usertype: "company representative", company_id: 1)
+    @rep.email_activate
+  end
+
+  describe "GET /foci --index" do
+    it "lets admin see all foci in the db" do
+      p '===============foci_spec.rb index 1'
+      p 'trying admin'
+      post "/login", :params=>{:user=>{:email=>'admin@admin.com',:password=>ENV["ADMIN_PW"]} }
+      parsed_body = JSON.parse(response.body)
+      expect(parsed_body["logged_in"]).to eq(true)
+      get '/foci'
+      expect(session["user_id"]).to eq(1)
+      expect(response).to have_http_status(:ok)
+      
+    end
+
+    it "lets student see all foci in the db" do
+      p '===============foci_spec.rb index 2'
+      p 'trying student'
+      post "/login", :params=>{:user=>{:email=>'js@student.com',:password=>'pw'} }
+      parsed_body = JSON.parse(response.body)
+      expect(parsed_body["logged_in"]).to eq(true)
+      get '/foci'
+      expect(session["user_id"]).to eq(3)
+      expect(response).to have_http_status(:ok)
+      
+    end
+
+    it "lets representative see all foci in the db" do
+      p '===============foci_spec.rb index 3'
+      p 'trying rep'
+      post "/login", :params=>{:user=>{:email=>'a@disney.com',:password=>'pw'} }
+      parsed_body = JSON.parse(response.body)
+      expect(parsed_body["logged_in"]).to eq(true)
+      get '/foci'
+      expect(session["user_id"]).to eq(2)
+      expect(response).to have_http_status(:ok)
+      
+    end
+
+    it "no data" do
+      p '===============foci_spec.rb index 4'
+      p 'trying admin'
+      Focus.destroy_all
+      post "/login", :params=>{:user=>{:email=>'admin@admin.com',:password=>ENV["ADMIN_PW"]} }
+      parsed_body = JSON.parse(response.body)
+      expect(parsed_body["logged_in"]).to eq(true)
+      get '/foci'
+      expect(session["user_id"]).to eq(1)
+      expect(response).to have_http_status(:ok)
+      parsed_body = JSON.parse(response.body)
+      expect(parsed_body["foci"]).to eq([])
+      
+    end
+
+    it "if not signed up and logged in no access" do
+      p '===============foci_spec.rb index 5'
```

Comment:
Remove all print statements

Reply:
print lines removed.

sorry but the whole interest model story is too big on pivotal tracker. should we divide it a little bit further?

Reply:
Viewing focuses shouldn't be locked behind having an account. Also, did you manually change the plural name to foci, or did Rails do that? Because I would rather it be focuses

Reply:
changed to focuses